### ISSUE TITLE
PSNR and SSIM test code

### ISSUE THREAD

**Issue by andycor** [HongguangZhang/DMPHN-cvpr19-master#8]
Can you provide PSNR and SSIM code ?
I never knew which function to use to calculate SSIM and PSNR. This problem has troubled me for a long time.
THX a lot.

**Comment by HongguangZhang** [comment]
I use Matlab built-in function ssim(img1, img2) and psnr(img1. img2) to calculate the them.

**Comment by andycor** [comment]
> I use Matlab built-in function ssim(img1, img2) and psnr(img1. img2) to calculate the them.

Thanks for your answer!!!
I have tried using matlab to calculate, but I found that the calculated SSIM is 0.91, which is different from the one in the paper, but the PSNR is the same.

**Comment by andycor** [comment]
Do you use the RGB channels to calculate SSIM? or use Y channel only?

**Comment by NoviceCrom** [comment]
I used compare_psnr imported from skimage.measure to calc the 1-2-4-8 pretrained model provided and I got PSNR 29.99 SSIM 0.8948 on test dataset.

**Comment by HongguangZhang** [comment]
Have you re-run test on all test images? The provided images in ./test_samples and ./test_results are just selected as a mini demo.

**Comment by NoviceCrom** [comment]
Yes,of course.1111 test pics in total.How many epochs have you run to obtain the 1-2-4-8 pretrained model?
Also,the average inference time in your 1-2-4-8 script is about 0.065s per pic,which is longer than
30s/1111=0.027s in the paper.

**Comment by hjbiao09** [comment]
@NoviceCrom Change deblur image and sharp image location in your SSIM code. Then u can get higher SSIM, which is similar to the paper.

**Comment by HongguangZhang** [comment]
> Yes,of course.1111 test pics in total.I ran 1-2-4-8 once again and got PSNR 30.45 SSIM 0.902.How many epochs have you run to obtain the 1-2-4-8 pretrained model?
> Also,the average inference time in your 1-2-4-8 script is about 0.065s per pic,which is longer than
> 30s/1111=0.027s in the paper.Confused.

Sorry for the delayed reply. 

I have no idea why you get a lower SSIM. As the provided models are re-trained, it is normal that there is small discrepancy on the PSNR and SSIM. But reducing from 0.93 to 0.90 is abnormal. 

Regarding the runtime, I used the P100 and V100 for evaluation. If not the same machine, the runtime will be different.

**Comment by agyptl** [comment]
> Yes,of course.1111 test pics in total.I ran 1-2-4-8 once again and got PSNR 30.45 SSIM 0.902.How many epochs have you run to obtain the 1-2-4-8 pretrained model?
> Also,the average inference time in your 1-2-4-8 script is about 0.065s per pic,which is longer than
> 30s/1111=0.027s in the paper.Confused.

Hi，I also meet this problem，the ssim is smaller. Have you solved this problem? Thanks

**Comment by agyptl** [comment]
> @NoviceCrom Change deblur image and sharp image location in your SSIM code. Then u can get higher SSIM, which is similar to the paper.

Excuse me, Would you like to elaborate on how to change the location? Thank you.

**Comment by hjbiao09** [comment]
@agyptl  I used python ssim code. If my memories are right., just switch the location. 
SSIM(deblur, GT) -> SSIM(GT, deblur). However i'm not sure...

**Comment by agyptl** [comment]
> @agyptl I used python ssim code. If my memories are right., just switch the location.
> SSIM(deblur, GT) -> SSIM(GT, deblur). However i'm not sure...

Thanks for your answer! I will take a try.

**Comment by agyptl** [comment]
> @agyptl I used python ssim code. If my memories are right., just switch the location.
> SSIM(deblur, GT) -> SSIM(GT, deblur). However i'm not sure...

Sorry! It doesn't work. Do you rember witch ssim function used, and use RGB channels average  number or Y  channel to calculate?
Thank you very much.

**Comment by YSL0226** [comment]
I got the same PSNR (30.45) and SSIM (0.902) which are exactly same as NoviceCrom got.
@HongguangZhang Could you confirm it? Because a lot of people got smaller SSIM.
